modules/core/04-channel/keeper/timeout.go

```diff
@@ -158,6 +158,11 @@ func (k Keeper) TimeoutExecuted(
 		k.SetChannel(ctx, packet.GetSourcePort(), packet.GetSourceChannel(), channel)
 	}
 
+	if channel.FlushStatus == types.FLUSHING && !k.hasInflightPackets(ctx, packet.GetSourcePort(), packet.GetSourceChannel()) {
+		channel.FlushStatus = types.FLUSHCOMPLETE
+		k.SetChannel(ctx, packet.GetSourcePort(), packet.GetSourceChannel(), channel)
+	}
+
```

Comment:
I think line 179 below just needs to be updated in line with the spec https://github.com/cosmos/ibc/pull/982/files#diff-6d4b263fc3c751e8e3fbd765b2817241db22c8e3083c61685900fa4783669220L1019

Reply:
just needed some slight re-ordering too